flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java

```diff
@@ -124,6 +124,9 @@ public void open(Configuration parameters) throws Exception {
 	@Override
 	public void run(SourceFunction.SourceContext<TimestampedFileInputSplit> context) throws Exception {
 		FileSystem fileSystem = FileSystem.get(new URI(path));
+		if (!fileSystem.exists(new Path(path))) {
+			throw new IOException("The provided file path " + path + " does not exist.");
```

Comment:
I think we should throw an `FileNotFoundException` which better reflects the reason of the error.